course5/homework12/line_sorting.cc

```diff
@@ -0,0 +1,47 @@
+#include <stdexcept>
+#include <limits>
+
+#include "line_sorting.h"
+
+namespace jds {
+
+std::vector<std::string> ReadLines(std::istream & source)
+{
+	std::vector<std::string> lines;
+	std::string line;
+	while (source)
+	{
+		std::getline(source, line);
+		if (bool(source) || line.length())
+			lines.push_back(std::move(line));
```

Comment:
Correct! You avoid deep copy of the string. But does it improve the overall performance of the function?